src/connectors/voice-search/__tests__/connectVoiceSearch-test.js

```diff
@@ -0,0 +1,188 @@
+import jsHelper, { SearchParameters } from 'algoliasearch-helper';
+import connectVoiceSearch from '../connectVoiceSearch';
+
+jest.mock('../../../lib/voiceSearchHelper', () => {
+  return ({ onStateChange, onQueryChange }) => {
+    return {
+      getState: () => {},
+      isBrowserSupported: () => true,
+      isListening: () => false,
+      toggleListening: () => {},
+      // ⬇️ for test
+      changeState: () => onStateChange(),
+      changeQuery: query => onQueryChange(query),
+    };
+  };
+});
+
+function getDefaultSetup() {
+  const renderFn = jest.fn();
+  const unmountFn = jest.fn();
+  const makeWidget = connectVoiceSearch(renderFn, unmountFn);
+  const widget = makeWidget({});
+  const helper = jsHelper({});
+
+  return {
+    renderFn,
+    unmountFn,
+    widget,
+    helper,
+  };
+}
+
+function getInitializedWidget() {
+  const { renderFn, unmountFn, widget, helper } = getDefaultSetup();
+
+  helper.search = () => {};
+  widget.init({ helper });
+
+  return {
+    renderFn,
+    unmountFn,
+    widget,
+    helper,
+    refine: query => widget._refine(query),
```

Comment:
Do we need to rely on the `_.refine()` internal? We should be able to get it from the rendering parameters.

Reply:
`connectVoiceSearch` doesn't provide `refine` as a rendering parameter to its component. All the state changes are happening in the instance of `voiceSearchHelper` and it's inside the connector. So it triggers refinement inside the connector.